app/src/models/branch.ts

```diff
@@ -68,6 +70,20 @@ export class Branch {
     return pieces[1]
   }
 
+  /** The name of remote for a remote branch. If local, will return null. */
+  public get remoteName(): string | null {
+    if (this.type === BranchType.Local) {
+      return null
+    }
+    const remoteRefPrefix = 'refs/remotes/'
+    const remoteBranchName = this.ref.replace(remoteRefPrefix, '')
+    const pieces = remoteBranchName.match(/(.*?)\/.*/)
+    if (!pieces || pieces.length < 2) {
+      return null
```

Comment:
I'm tempted to say this should throw an error instead of returning `null`. If it's a remote branch, and it has a `ref` (which it should), it should conform to the expected format (`/refs/remotes/<remote-name>/<branch-name>`), otherwise bad things could happen so we better abort the operation and crash / display an error.
```suggestion
      throw new Error(`Remote branch ref has unexpected format: ${this.ref}`)
```